### DISCUSSION TITLE
C# Determine when the line has been crossed

### DISCUSSION THREAD

**Discussion by peacemaker79** [emgucv/emgucv#970] [Q&A]
hello everyone
At the moment I am using Opencv and Absdiff to determine when the line has been crossed. The line can be any color or even laser red. Is my idea good or am I on the wrong way.
The screenshot is only for viewing the camera direction.
Many Thanks for suggestions.
![Screenshot_20241105-193923~2](https://github.com/user-attachments/assets/ea158e8a-667a-4328-9f86-2c9241acf2ea)

**Comment by emgucv** [comment]
Your shadow can change the color of the line. 

If you can place two QR Code at both ends of the line. If your camera have sufficient resolution to detect the QR code, use the Open CV QRCode detector to detect the QRCode and draw a virtual line between the two:

https://github.com/emgucv/emgucv/blob/e4f99cdbaf6c87bd28861b459b0a7e18d861429b/Emgu.CV.Models/Dnn/WeChatQRCodeDetector.cs#L186

That should give you a more accurate line location. 

You can use motion detector to get a motion mask (pixels with motions), if that over laps the line formed by the QR code, there is motion on the line.

https://www.emgu.com/wiki/files/4.9.0/document/html/T_Emgu_CV_MotionHistory.htm
